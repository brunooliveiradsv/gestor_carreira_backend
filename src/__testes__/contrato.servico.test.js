// src/__testes__/contrato.servico.test.js
const PDFDocument = require('pdfkit');
const contratoServico = require('../servicos/contrato.servico');
const { Writable } = require('stream');

// --- SIMULAÇÃO (MOCK) ATUALIZADA DO PDFKIT ---
let capturedText = '';
const mockPdfDoc = {
  pipe: jest.fn(),
  fontSize: jest.fn().mockReturnThis(),
  font: jest.fn().mockReturnThis(),
  lineGap: jest.fn().mockReturnThis(),
  text: jest.fn((text, options) => {
    capturedText += `${text}\n`;
    return mockPdfDoc;
  }),
  moveDown: jest.fn().mockReturnThis(),
  list: jest.fn((items) => {
    items.forEach(item => {
      capturedText += `- ${item}\n`;
    });
    return mockPdfDoc;
  }),
  end: jest.fn(),
  // --- CORREÇÃO AQUI ---
  // Adicionamos as propriedades em falta com valores fictícios para os cálculos de layout
  y: 700, // Um valor fictício para a posição vertical (doc.y)
  page: {   // Um objeto fictício para a página (doc.page)
    width: 612, // Largura padrão de uma página A4 em pontos
    margins: {
      left: 72,
      right: 72,
    },
  },
};

jest.mock('pdfkit', () => {
  return jest.fn().mockImplementation(() => mockPdfDoc);
});

describe('Testes do Serviço de Geração de Contratos', () => {

  beforeEach(() => {
    capturedText = '';
    jest.clearAllMocks();
  });

  it('Deve gerar o texto do contrato corretamente', () => {
    // 1. Criar dados de teste (mocks)
    const compromissoMock = {
      nome_evento: 'Show Acústico na Praça',
      data: new Date('2025-12-25T20:00:00.000Z'),
      local: 'Praça da Matriz, Mirassol',
      valor_cache: 1500.75,
    };
    const contratanteMock = {
      nome: 'Prefeitura de Mirassol',
      nif: '12.345.678/0001-99',
      morada: 'Rua Principal, 123, Centro',
      forma_pagamento: '50% adiantado, 50% no dia do evento.',
      cidade_foro: 'Mirassol',
      estado_foro: 'SP',
    };
    const artistaMock = {
      nome: 'Bruno Oliveira',
    };
    const streamMock = new Writable({
        write(chunk, encoding, callback) {
          callback();
        }
    });

    // 2. Chamar a função que queremos testar
    contratoServico.gerarContratoPDF(compromissoMock, contratanteMock, artistaMock, streamMock);

    // 3. Usar toMatchSnapshot()
    expect(capturedText).toMatchSnapshot();
  });
});