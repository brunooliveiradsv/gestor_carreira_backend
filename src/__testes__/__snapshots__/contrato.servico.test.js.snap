// Jest Snapshot v1, https://jestjs.io/docs/snapshot-testing

exports[`Testes do Serviço de Geração de Contratos Deve gerar o texto do contrato corretamente 1`] = `
"CONTRATO DE PRESTAÇÃO DE SERVIÇOS ARTÍSTICOS
DAS PARTES
CONTRATANTE: Prefeitura de Mirassol, portador(a) do CPF/CNPJ nº 12.345.678/0001-99, com morada em Rua Principal, 123, Centro, doravante designado(a) simplesmente como CONTRATANTE.
CONTRATADO: Bruno Oliveira, músico(a) profissional, doravante designado(a) simplesmente como CONTRATADO.
Cláusula 1ªª - DO OBJETO
O presente contrato tem como objeto a apresentação musical do CONTRATADO no evento "Show Acústico na Praça", a ser realizado no dia 25 de dezembro de 2025.
Cláusula 2ªª - DO LOCAL E HORÁRIO
A apresentação ocorrerá no seguinte local: Praça da Matriz, Mirassol. O CONTRATADO deverá estar disponível no local com, no mínimo, 60 (sessenta) minutos de antecedência. A apresentação terá início previsto para as 17:00h.
Cláusula 3ªª - DAS OBRIGAÇÕES DO CONTRATADO
Compete ao CONTRATADO:
- Realizar a apresentação musical conforme acordado, com duração aproximada a ser definida entre as partes.
- Comparecer ao local do evento no horário estipulado, munido de seus instrumentos e equipamentos pessoais necessários à apresentação.
- Zelar pela qualidade técnica e artística da apresentação.
Cláusula 4ªª - DAS OBRIGAÇÕES DO CONTRATANTE
Compete ao CONTRATANTE:
- Disponibilizar um local seguro e adequado para o CONTRATADO se apresentar.
- Garantir a segurança do CONTRATADO e de seus equipamentos durante todo o período de sua permanência no local do evento.
- Efetuar o pagamento do cachê nos termos da Cláusula 5ª.
Cláusula 5ªª - DO PAGAMENTO
A título de remuneração (cachê), o CONTRATANTE pagará ao CONTRATADO o valor total de R$ 1.500,75. O pagamento deverá ser efetuado da seguinte forma: 50% adiantado, 50% no dia do evento..
Cláusula 6ªª - DA RESCISÃO
O cancelamento por qualquer uma das partes deverá ser comunicado com antecedência mínima de 15 (quinze) dias. O descumprimento desta cláusula por parte do CONTRATANTE implicará no pagamento de uma multa de 50% do valor total do cachê. Caso o cancelamento parta do CONTRATADO, este deverá restituir qualquer valor já recebido.
Cláusula 7ªª - DAS DISPOSIÇÕES GERAIS
O CONTRATANTE autoriza o uso de imagem e som da apresentação para fins de divulgação e portfólio do CONTRATADO. Este contrato não estabelece qualquer vínculo empregatício entre as partes, tratando-se de uma prestação de serviço autônoma.
Cláusula 8ªª - DO FORO
Fica eleito o foro da comarca de Mirassol, SP, para dirimir quaisquer dúvidas oriundas do presente contrato.
Mirassol, 25 de julho de 2025.
________________________________
________________________________
Prefeitura de Mirassol
Bruno Oliveira
CONTRATANTE
CONTRATADO
"
`;
